corehq/motech/requests.py

```diff
@@ -292,3 +311,42 @@ def validate_user_input_url_for_repeaters(url, domain, src):
                 'reason': e.reason,
             })
             raise
+
+
+def apply_request_backoff(
+    max_retries=MAX_TRIES,
+    exponential_base=DEFAULT_BACKOFF_RETRY_EXPONENT_BASE,
+    exponential_period=DEFAULT_EXPONENTIAL_PERIOD
+):
+    def request_handler(request_function):
+        def make_request_with_backoff(*args):
+            retry_count = 0
+            response = None
+
+            while True:
+                try:
+                    response = request_function(*args)
+                except (RequestException, HTTPError):
+                    if retry_count < max_retries:
+                        backoff_period_seconds = _get_exponential_backoff_period_in_seconds(retry_count)
+                        sleep(backoff_period_seconds)
```

Comment:
Are there better ways than to use the sleep() function? This is running in it's own thread so I don't foresee any issues, but just wondering.